website/blog/2023-08-08-deploy-run-ferretdb-leafcloud.md

```diff
@@ -0,0 +1,443 @@
+
+### Creating a Kubernetes Cluster in Leafcloud
+
+We will start by creating a Kubernetes cluster, using the OpenStack CLI.
+Leafcloud manages and deploys container clusters using the OpenStack Magnum project.
+
+Start by installing the OpenStack CLI using the following command:
+
+```sh
+sudo apt update -y
+sudo apt install -y python3-pip python3-dev -y
+sudo apt install virtualenv -y
+virtualenv -p python3 openstack_venv
+source openstack_venv/bin/activate
+pip install --upgrade pip
+pip install python-openstackclient
+pip install python-magnumclient
+```
+
+Once it's installed, you need to log in to your Leafcloud account to download the OpenStack RC file that contains the environment variables for your command-line client.
+
+After downloading the file, copy and paste the file's contents into a new document as:
+
+```text
+~/leafcloudopenrc.sh
+```
+
+To set up the configuration for your Leafcloud account, run the following command:
+
+```sh
+source ~/leafcloudopenrc.sh
+```
+
+It will prompt you to enter your account password.
+When you do, you will have access to the OpenStack CLI for your Leafcloud account.
+
+To check, run the command below to see if you can access the server list using the OpenStack CLI:
+
+```sh
+openstack server list
+```
+
+If you get an authentication error, try to run `source ~/leafcloudopenrc.sh` again and enter the right password for your account.
+
+Leafcloud provides a set of cluster templates that you can take advantage of right away.
+See the templates by running:
+
+```sh
+openstack coe cluster template list
+```
+
+Select the best template for your project.
+
+We'll be using the `K8s-ha-v1.21.2-template-v2.0-rc3` template for the guide since it comes with Kubernetes OpenStack autoscaling, encrypted cinder volumes for containers and hosts (persistent volume claims), and high-availability load balancers.
+
+Using the selected template, we can now create a new cluster.
+The keypair, which we associate using the `--keypair` parameter, will be integrated into the hosts, granting us root SSH access (with 'core' as the default user).
+The keypairs are essential for encrypting SSH traffic for your instances and enabling secure communications.
+
+To create your cluster, use this command:
+
+```sh
+openstack coe cluster create my-k8s-cluster --cluster-template k8s-ha-v1.21.2-template-v2.0-rc3 --keypair <keypair>
+```
+
+You may need to wait a few minutes for this process to be complete.
+To check up on the installation enter the following:
+
+```sh
+openstack coe cluster list
+```
+
+The installation is complete once the status of the cluster changes from CREATE_IN_PROGRESS to CREATE_COMPLETE.
+
+Now we need to fetch the configuration file for the cluster.
+Do this by running the following:
+
+```sh
+openstack coe cluster config my-k8s-cluster
+```
+
+A file named 'config' will be downloaded to your home directory.
+
+```sh
+export KUBECONFIG=/home/<username>/config
```

Comment:
This command sets  `KUBECONFIG` env var to the given value,  but there is not explanation why we do it. If I understand correctly, it's because below we always run `kubectl` with no config being provided. It would be nice to explain that we want to set the default KUBECONFIG here.